runtime/bindings/python/hal.cc

```diff
@@ -312,11 +313,21 @@ py::list HalDriver::QueryAvailableDevices() {
   return results;
 }
 
+// Configures |device| based on flags before returning it to the user.
+static iree_status_t ConfigureDevice(iree_hal_device_t* device) {
+  // Optionally wrap the base device allocator with caching/pooling.
+  // Doing this here satisfies the requirement that no buffers have been
+  // allocated yet - if we returned the device without doing this the caller can
+  // more easily break the rules.
+  return iree_hal_configure_allocator_from_flags(device);
```

Comment:
How do flags interface with runtime/bindings/python?

Reply:
There's a method exposed for parsing flags so the python side can pass through whatever it wants and update the flag states: https://github.com/openxla/iree/blob/main/runtime/bindings/python/tests/flags_test.py#L16
(currently used for setting task system worker count and such as well as cuda/vulkan device options)

Reply:
Ah, so that updates the global flags? Seems sketchy vs a --extra_args / kwargs field on a single function call.

Reply:
🤷 this is what it does, and this PR doesn't change it

Reply:
(the previous PRs added the programmatic access for string-based configuration so that we could switch to using that if we wanted, this PR just wraps that and makes it turn-key, future bindings authors can decide what they want to do)

Reply:
There may be some broken assumptions here, though - I'm going to dig a bit separate from this PR.

Reply:
I worry about having multiple global flags that users set between calls, compared to having something stateful that gets attached to a specific python function call / object. This flag on its own doesn't seem _that_ risky, but we might want different ergonomics for such configurations through the Python bindings. If plumbing options and flags through layers of Python/pybind/C++/C is tedious, would it help to add a JSON wrapper (or config protobuf :P), for example?

Reply:
yeah, and I think there may be some bugs we've been skating by with this usage today - it's fine for ints and such but not strings. I'm taking a look at how to either fix that case or fix this usage.